docs/guides/blocks/actions.md

```diff
@@ -2,6 +2,85 @@
 title: Actions
 ---
 
-# Add custom actions
+# Add custom actions for buttons and automations
 
-TODO
+In Coda users can take actions directly within their documents, using native [buttons][help_buttons] and [automations][help_automations]. Packs can provide custom actions which users can use when building out their doc. These actions often use the [fetcher][fetcher] to send data to an external API, but they can also be used for other one-time calculations.
+
+[View Sample Code][samples]{ .md-button }
+
+
+## Using actions
+
+Actions provided by Packs appear as choices in the button and automation dialogs along-side built-in actions. Actions can also be used in the formula editor for those buttons and automations.
+
+=== "In a button"
+    <img src="../../../images/actions_button.png" srcset="../../../images/actions_button_2x.png 2x" class="screenshot" alt="Selecting an action for a button">
+=== "In an automation"
+    <img src="../../../images/actions_automation.png" srcset="../../../images/actions_automation_2x.png 2x" class="screenshot" alt="Selecting an action for an automation">
+=== "In the formula editor"
+    <img src="../../../images/actions_formula.png" srcset="../../../images/actions_formula_2x.png 2x" class="screenshot" alt="Using actions in the formula editor.">
+
+
+## Creating actions
+
+Actions are just a special type of formula, and to create an action simply add a formula to your Pack with the property `isAction: true`.
+
+```ts
+pack.addFormula({
+  name: "DoSomething",
+  description: "Does something neat.",
+  isAction: true,
+  execute: async function ([], context) {
+    // ...
+  },
+});
+```
+
+If you aren't already familiar with creating formulas, read the [Formulas guide][formulas] first. The rest of this guide will clarify how action formulas differ from calculation formulas.
+
+
+## Parameters
+
+Parameters in actions work the same as they do in other formulas, but may appear differently depending on the context. The button and automation configuration dialogs by default will present the parameters as input boxes, although the user has the option to switch the formula editor as well. See the [Parameters guide][parameters] for more information and examples.
+
+
+## Results
+
+Actions can return a value but aren't required to. When the action is used by a button in a table column, the result can be stored in another column of the same table. For this reason it's a best practice to return some value, so that other formulas or conditional formatting can check that column and know when the action is complete. For example, if your action is creating an item in an external API you could return the URL or ID of the new item.
```

Comment:
Per Slack convo, would suggest also adding a line that if there isn't a meaningful unique value to return, consider just returning "OK" as an indicator that the action was completed successfully.

Reply:
Done. PTAL.